### ISSUE TITLE
Layers adding to map twice (sometimes)

### ISSUE THREAD

**Issue by philipbaileynar** [Riverscapes/ArcProViewer#1]
Sometimes, and I know what circumstances cause this, but layers get added to the map twice. This can happen when double clicking the project tree or right clicking and using the context menu. It also happens for rasters, vectors and WMS.

<img width="321" alt="Screenshot 2024-07-08 at 8 44 08 AM" src="https://github.com/Riverscapes/ArcProViewer/assets/8645379/34c865fa-91b3-4e9c-863a-f2e88a77cf43">

**Comment by philipbaileynar** [comment]
Zach experienced this at least once in his testing.

**Comment by philipbaileynar** [comment]
I solved this today. The problem was two fold:

1. When looking for existing Group Layers in the map ToC I was doing this by layer name. ArcPro has a `FinderLayers()` method that takes a special map ToC URi to safely look for existing layers. I was already using this approach for individual data layers and simply extended it to work for Group Layers as well.
2. Expanding Group Layers immediately after adding them seem to be the cause of the ToC ghost layers. So I moved the expanding Group Layers to the end of the process after all the additions are complete.
